### PR TITLE
Added idiotic skin option for shiny button lovers

### PR DESCRIPTION
**Pull request by frozsgy**
A skin implementation for people who likes that kind of thing. Heavily inspired from a badly designed website.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -616,3 +622,67 @@ function prevb(){
...
+
+function idiot(){
+	rmcourses();
+	rmblocks();
+	courses = [];
+	dontfill_color="#000";
+	palette = ["#CC0000", "#7A00CC", "#29A329", "#CCCC00",  "#00CCCC", "#00008A", "#002900", "#E62EB8", "#005C5C", "#CC3300", "#808080", "#00FF00", "#666633", "#002E2E"];
+	var style = document.createElement('style');
+	style.innerHTML =
+	'.courses { background-color: #e44424 !important; color: #fff !important; }' +
```
Comment (duck2) [inline main.js]: Maybe it's better to add an `idiot` class to every element on the page (`document.querySelectorAll("body *")`), and have this styling separately in CSS.

```
.ccode.idiot { ... }
.block.idiot { ... }
```

File: main.js [hunk 2]
```diff
@@ -20,6 +20,10 @@ var dontfills = [], dontfill_color="#ddd";
  * "unbound": the current schedule is not possible with current options.
  * it won't be updated until a successful make() call. */
 var state = "blank";
+
+/* keeps track of if user is idioting */
+var iq = 1;
```
Comment (duck2) [inline main.js]: eh. nice burn but better to have it as a positive switch `var idiot = 0;`

File: main.js (hunk 1 above)
Comment (frozsgy) [reply main.js]: working on it, thanks for the feedback!

File: main.js [hunk 3]
```diff
@@ -629,10 +634,12 @@ function rmcourses(){
 	courses = [];
 }
 
-function idiot(){
-	rmcourses();
-	rmblocks();
-	courses = [];
+function idiot(isLoad = false){
+	if(!isLoad) {
```
Comment (duck2) [inline main.js:643]: why is this?

Comment (frozsgy) [reply main.js:643]: I forgot to finish implementing the skin change function properly apparently. Now it should make sense, with the latest commit.

File: main.js [hunk 4]
```diff
@@ -616,3 +627,88 @@ function prevb(){
...
+		rmcourses();
+		rmblocks();
+		courses = [];
+	}
+	dontfill_color = "#000";
+	palette = ["#CC0000", "#7A00CC", "#29A329", "#CCCC00",  "#00CCCC", "#00008A", "#002900", "#E62EB8", "#005C5C", "#CC3300", "#808080", "#00FF00", "#666633", "#002E2E"];
+	var elButtons = document.querySelectorAll('button');
+	for(i = 0; i < elButtons.length; i++) {
+  		elButtons[i].classList.add('button-idiot');
```
Comment (duck2) [inline main.js]: I would think something like
```js
all = document.querySelectorAll("*");
for(i=0; i<all.length; i++) all[i].classList.add("idiot");
```
would work better? Then you can style with
```css
.courses.idiot {
    background-color...
}
```

Comment (frozsgy) [reply main.js]: Thanks for the input, now it's way better!

File: main.js [hunk 5]
```diff
@@ -616,3 +627,60 @@ function prevb(){
...
+		all[i].classList.remove("idiot");
+	}
+	isidiot = 0;
+	document.getElementById('idiot-link').innerHTML = 'i am an idiot!';
+	restorestate(tempstate);
+}
+
+grab("idiot-link").onclick = function(){ if(isidiot === 1) { clever();} else { idiot();} };
+
+function idiotCheck() {
```
Comment (duck2) [inline main.js]: eh I missed this. the right way to do this is via a [CSS media query](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries#Targeting_media_features). don't touch the JS, go in the CSS:
```css
@media(max-width: 768px){
    #idiot-link { display: none }
}
```

File: index.html [hunk 6]
```diff
@@ -5,11 +5,13 @@
 <link rel="icon" href="favicon.ico" type="image/x-icon">
 <link rel="stylesheet" href="style.css">
 <link rel="stylesheet" href="awesomplete.css">
+<body onLoad="idiotCheck();">
 <div class="header">
 <a id="index">tetick.xyz</a>
 <a id="save">save</a>
 <a id="about-link">about</a>
 <a id="wrong-link">wrong!</a>
+<a id="idiot-link" title="who likes shiny big buttons with disgusting colours">i am an idiot!</a>
```
Comment (duck2) [inline index.html]: I think the link can be "ugly!" and the alt title "i am an idiot who likes..." so it looks _friendlier_.

File: main.js (hunk 5 above)
Comment (frozsgy) [reply main.js]: Done, thanks for the info. I didn't know about that media queries ^^

**Comment by duck2** [comment]
ok iguess

**Pull request merged**